api/v1beta1/azuremachine_webhook.go

```diff
@@ -213,6 +213,13 @@ func (mw *azureMachineWebhook) ValidateUpdate(ctx context.Context, oldObj, newOb
 		allErrs = append(allErrs, err)
 	}
 
+	if err := webhookutils.ValidateImmutable(
+		field.NewPath("spec", "disableExtensionOperations"),
```

Comment:
```suggestion
		field.NewPath("spec", "DisableExtensionOperations"),
```

This should also be fixed for `capacityReservationGroupID` on line 210. Sorry for not catching this in the previous PR!

Reply:
Changed to use the capitalized variant. `capacityReservationGroupID` should be fixed in its own PR.

Reply:
Why with a capital? This breaks Kube API conventions. The path is supposed to take the json serialisation representation of the field name, so a `camelCase` is correct.

Adding `DisableExtensionOperations` with a capital means the error that comes back prints an incorrect JSON path

The `capacityReservationGroupID` is correct and should not be changed

Reply:
Hmm you're definitely correct. I'm seeing some inconsistencies in the use of `field.NewPath()` in CAPZ and that should definitely be fixed. Apologies for the confusion there!

Reply:
This PR is changed back to camelCase.